Realm/RLMDictionary.h

```diff
@@ -0,0 +1,442 @@
+
+/**
+ Returns a distinct `RLMResults` from the dictionary.
+
+ @param keyPaths     The key paths to distinct on.
+
+ @return    An `RLMResults` with the distinct values of the keypath(s).
+ */
+- (RLMResults<RLMObjectType> *)distinctResultsUsingKeyPaths:(NSArray<NSString *> *)keyPaths;
+
+#pragma mark - Aggregating Property Values
+
+/**
+ Returns the minimum (lowest) value of the given property among all the values in the dictionary.
+
+     NSNumber *min = [object.dictionaryProperty minOfProperty:@"age"];
+
+ @warning You cannot use this method on `RLMObject`, `RLMDictionary`, and `NSData` properties.
+
+ @param property The property whose minimum value is desired. Only properties of
+                 types `int`, `float`, `double`, and `NSDate` are supported.
+
+ @return The minimum value of the property, or `nil` if the dictionary is empty.
+ */
+- (nullable id)minOfProperty:(NSString *)property;
+
+/**
+ Returns the maximum (highest) value of the given property among all the objects in the dictionary.
+
+     NSNumber *max = [object.dictionaryProperty maxOfProperty:@"age"];
+
+ @warning You cannot use this method on `RLMObject`, `RLMDictionary`, and `NSData` properties.
+
+ @param property The property whose maximum value is desired. Only properties of
+                 types `int`, `float`, `double`, and `NSDate` are supported.
+
+ @return The maximum value of the property, or `nil` if the dictionary is empty.
+ */
+- (nullable id)maxOfProperty:(NSString *)property;
+
+/**
+ Returns the sum of distinct values of a given property over all the objects in the dictionary.
+
+     NSNumber *sum = [object.dictionaryProperty sumOfProperty:@"age"];
+
+ @warning You cannot use this method on `RLMObject`, `RLMDictionary`, and `NSData` properties.
+
+ @param property The property whose values should be summed. Only properties of
+                 types `int`, `float`, and `double` are supported.
+
+ @return The sum of the given property.
+ */
+- (NSNumber *)sumOfProperty:(NSString *)property;
+
+/**
+ Returns the average value of a given property over the objects in the dictionary.
+
+     NSNumber *average = [object.dictionaryProperty averageOfProperty:@"age"];
+
+ @warning You cannot use this method on `RLMObject`, `RLMDictionary`, and `NSData` properties.
+
+ @param property The property whose average value should be calculated. Only
+                 properties of types `int`, `float`, and `double` are supported.
+
+ @return The average value of the given property, or `nil` if the dictionary is empty.
+ */
+- (nullable NSNumber *)averageOfProperty:(NSString *)property;
+
+#pragma mark - Notifications
+
+/**
+ Registers a block to be called each time the dictionary changes.
+
+ The block will be asynchronously called with the initial dictionary, and then
+ called again after each write transaction which changes any of the key-value in
+ the dictionary or which objects are in the results.
```

Comment:
This looks like an incorrect find/replace on the docs from the array version and isn't a grammatically correct sentence or an accurate description of the behavior.